imodels/util/score.py

```diff
@@ -63,19 +64,32 @@ def _eval_rule_perf(rule, X, y) -> Tuple[float, float]:
     return y_detected.mean(), float(true_pos) / pos
 
 
-def score_lasso(X, y, rules: List[str], Cs, cv, random_state) -> Tuple[List[Rule], LassoCV]:
-    if Cs is None:
-        n_alphas = 100
-        alphas = None
-    elif hasattr(Cs, "__len__"):
-        n_alphas = None
-        alphas = 1. / Cs
-    else:
-        n_alphas = Cs
-        alphas = None
-    lscv = LassoCV(n_alphas=n_alphas, alphas=alphas, cv=cv, random_state=random_state)
+def score_lasso(X, y, rules: List[str], alphas=None, cv=3, max_rules=2000, random_state=None) -> Tuple[List[Rule], Lasso]:
+    if alphas is None:
+        alphas = _alpha_grid(X, y) 
+
+    mse_scores = []
+    nonzero_rule_coefs_count = []
+    kf = KFold(cv)
+    for alpha in alphas:
+        m = Lasso(alpha=alpha, random_state=random_state)
+        mse = 0
+        for train_index, test_index in kf.split(X):
+            X_train, X_test = X[train_index], X[test_index]
+            y_train, y_test = y[train_index], y[test_index]
+            m.fit(X_train, y_train)
+            mse += np.mean((m.predict(X_test) - y_test) ** 2)
+        
+        rule_count = sum(np.abs(m.coef_[-len(rules):]) > 0.001)
```

Comment:
This uses 0.001 as the threshold for when we consider a coef to be basically zero — lmk if it should just be 0

Reply:
Good call, this shouldn't be exactly zero but we may want to tolerate small coefs when the values in y are themselves small or there is a big imbalance. Could you make the threshold `1e-6 / np.mean(np.abs(y_train)` ?

Reply:
Done